HPCC-14389 ProcessPerSlave change and thor related configxml cleanup

Michael-Gardner opened the pull request:
Signed-off-by: Michael Gardner michael.gardner@lexisnexis.com

Michael-Gardner committed:
HPCC-14389 ProcessPerSlave change and thor related configxml cleanup

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Michael-Gardner commented:
@jakesmith @garonsky Please review when you have time

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented on initfiles/componentfiles/configxml/setvars_linux.xsl:
```diff
@@ -42,108 +35,66 @@
+                     </xsl:call-template>
 export autoSwapNode=<xsl:choose>
            <xsl:when test="SwapNode/@AutoSwapNode='true'">1</xsl:when>
            <xsl:when test="SwapNode/@AutoSwapNode='1'">1</xsl:when>
```
Not new, but don't see why this property in particular vs others is checking '1' ,

jakesmith commented:
Looks a lot cleaner.
@garonsky , please double check

Michael-Gardner committed:
HPCC-14389 Added boolean check to autoSwapNode

Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

Michael-Gardner commented:
@jakesmith I've gone ahead and modified the autoSwapNode code to do a boolean() check on the elements value, and then output true or false.  I've gone ahead and changed the appropriate code in init_thor to make this change work.  All the c++ code referencing AutoSwapNode is using getBoolean so we should be fine there.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

Michael-Gardner commented:
@garonsky can you please give the xslt a glance before I @ Richard?

garonsky commented:
Looks good.

Michael-Gardner commented:
@richardkchapman Ready for a merge :+1:
